ops/charm.py

```diff
@@ -371,9 +371,10 @@ class RelationEvent(HookEvent):
     relation: 'Relation'
     """The relation involved in this event."""
 
-    # TODO(benhoyt): I *think* app should never be None, but confirm and update type
     app: Optional[model.Application]
-    """The remote application that has triggered this event."""
+    """The remote application that has triggered this event.
+
+    This may be ``None`` for relation-broken events."""
```

Comment:
I think we should just bite the bullet and make this `app: model.Application` (non-optional) for all of these. It seems like from all accounts it should be set, and it's a pretty big Juju issue if it's not (the relation-broken issue some have seen).

Minor style nit: let's put the ending `"""` on its own new line -- not enforced by linting, but that's the style we've used elsewhere for multi-line docstrings, and I think it makes the two `"""`'s line up nicely and marks a bit more clearly where the block comment ends.